File: Advanced/GuiDemo/app/src/main/java/edu/cs4730/guidemo/Picker_Fragment.java
package edu.cs4730.guidemo;

import android.app.DatePickerDialog;
import android.app.TimePickerDialog;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.DatePicker;
import android.widget.TimePicker;

import java.text.DateFormat;
import java.util.Calendar;

import androidx.fragment.app.Fragment;

import edu.cs4730.guidemo.databinding.PickerFragmentBinding;

/**
 * shows how to use pickers.
 */
public class Picker_Fragment extends Fragment implements Button.OnClickListener {
    //used for the pickers to set the current time/date
    PickerFragmentBinding binding;
    Calendar dateAndTime = Calendar.getInstance();
    DateFormat fmtDate = DateFormat.getDateInstance();
    DateFormat fmtTime = DateFormat.getTimeInstance();
    //listeners for the data and time picker.
    DatePickerDialog.OnDateSetListener d;
    TimePickerDialog.OnTimeSetListener t;

    public Picker_Fragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = PickerFragmentBinding.inflate(inflater, container, false);
        binding.btnDate.setOnClickListener(this);
        binding.btnTime.setOnClickListener(this);

        //create the date picker listener, which is used later when the picker is called.
        d = new DatePickerDialog.OnDateSetListener() {
            public void onDateSet(DatePicker view, int year, int monthOfYear, int dayOfMonth) {
                dateAndTime.set(Calendar.YEAR, year);
                dateAndTime.set(Calendar.MONTH, monthOfYear);
                dateAndTime.set(Calendar.DAY_OF_MONTH, dayOfMonth);
                binding.tvDate.setText(fmtDate.format(dateAndTime.getTime()));
            }
        };
        //create the time picker listener
        t = new TimePickerDialog.OnTimeSetListener() {
            public void onTimeSet(TimePicker view, int selectedHour, int selectedMinute) {
                dateAndTime.set(Calendar.HOUR, selectedHour);
                dateAndTime.set(Calendar.MINUTE, selectedMinute);
                // set current time into textview
                binding.tvTime.setText(fmtTime.format(dateAndTime.getTime()));
            }
        };
        return binding.getRoot();
    }

    @Override
    public void onClick(View v) {

        if (v == binding.btnDate) {//date picker
            new DatePickerDialog(requireContext(), d,
                    dateAndTime.get(Calendar.YEAR),
                    dateAndTime.get(Calendar.MONTH),
                    dateAndTime.get(Calendar.DAY_OF_MONTH)
            ).show();
        } else { //time picker
            Log.v("Time", "Should show the picker!");
            new TimePickerDialog(requireActivity(), t,
                    dateAndTime.get(Calendar.HOUR),
                    dateAndTime.get(Calendar.MINUTE),
                    false
            ).show();
        }
    }
}

File: Advanced/BotNavGuiDemo_kt/app/src/main/java/edu/cs4730/botnavguidemo_kt/Picker_Fragment.kt
package edu.cs4730.botnavguidemo_kt

import android.app.DatePickerDialog
import android.app.DatePickerDialog.OnDateSetListener
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.*
import androidx.fragment.app.Fragment
import edu.cs4730.botnavguidemo_kt.databinding.PickerFragmentBinding
import java.text.DateFormat
import java.util.Calendar

/**
 * example of pickers.
 */
class Picker_Fragment : Fragment(), View.OnClickListener {
    lateinit var binding: PickerFragmentBinding

    //used for the pickers to set the current time/date
    var dateAndTime = Calendar.getInstance()
    var fmtDate = DateFormat.getDateInstance()
    var fmtTime = DateFormat.getTimeInstance()

    //listeners for the data and time picker.
    lateinit var d: OnDateSetListener
    lateinit var t: OnTimeSetListener
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        // Inflate the layout for this fragment

        // Inflate the layout for this fragment
        binding = PickerFragmentBinding.inflate(inflater, container, false)
        binding.btnDate.setOnClickListener(this)
        binding.btnTime.setOnClickListener(this)

        //create the date picker listener, which is used later when the picker is called.
        d = OnDateSetListener { view, year, monthOfYear, dayOfMonth ->
            dateAndTime[Calendar.YEAR] = year
            dateAndTime[Calendar.MONTH] = monthOfYear
            dateAndTime[Calendar.DAY_OF_MONTH] = dayOfMonth
            binding.tvDate.text = fmtDate.format(dateAndTime.time)
        }
        //create the time picker listener
        t = OnTimeSetListener { view, selectedHour, selectedMinute ->
            dateAndTime[Calendar.HOUR] = selectedHour
            dateAndTime[Calendar.MINUTE] = selectedMinute
            // set current time into textview
            binding.tvTime.text = fmtTime.format(dateAndTime.time)
        }
        return binding.root
    }

    override fun onClick(v: View) {
        if (v === binding.btnDate) { //date picker
            DatePickerDialog(
                requireActivity(), d,
                dateAndTime[Calendar.YEAR],
                dateAndTime[Calendar.MONTH],
                dateAndTime[Calendar.DAY_OF_MONTH]
            ).show()
        } else { //time picker
            Log.v("Time", "Should show the picker!")
            TimePickerDialog(
                activity, t,
                dateAndTime[Calendar.HOUR],
                dateAndTime[Calendar.MINUTE],
                false
            ).show()
        }
    }
}